### PR TITLE
Clarify HOSTPATH in docker cp

### LINKED ISSUE #11141: docker cp: notion of HOSTPATH
```
$ docker cp --help

Usage: docker cp [OPTIONS] CONTAINER:PATH HOSTPATH

Copy files/folders from the PATH to the HOSTPATH

  --help=false       Print usage
```

This usage of HOSTPATH always strikes me as it's actually copying and extracting the files over where the Docker daemon host lives. (Isn't that what we usually refer to as "host"?) This sort of confuses me and there's not really a way to find out without trying out or reading more docs on docker.com. 

I'd love to propose rewriting this usage string with a less confusing one. Opinions?

### PR DESCRIPTION
**Pull request by ahmetb**
Fixes #11141. It wasn't really clear if HOSTPATH is referring the daemon host or cli host.

cc: @duglin @cpuguy83 
Signed-off-by: Ahmet Alp Balkan ahmetalpbalkan@gmail.com

### CODE REVIEW

**Comment by jfrazelle** [comment]
LGTM

File: docs/man/docker-cp.1.md [hunk 1]
```diff
@@ -11,8 +12,9 @@ CONTAINER:PATH HOSTPATH
 
 # DESCRIPTION
 Copy files/folders from a container's filesystem to the host
-path. Paths are relative to the root of the filesystem. Files
-can be copied from a running or stopped container.
+path running the command. Paths are relative to the root of
```
Comment (duglin) [inline docs/man/docker-cp.1.md]: only the path in the container is relative to the root - the HOSTPATH is relative to the current dir (unless it starts with /)

File: docs/sources/reference/commandline/cli.md [hunk 2]
```diff
@@ -742,11 +742,12 @@ Supported `Dockerfile` instructions: `CMD`, `ENTRYPOINT`, `ENV`, `EXPOSE`,
 ## cp
 
 Copy files/folders from a container's filesystem to the host
-path.  Paths are relative to the root of the filesystem.
+path. Paths are relative to the root of the filesystem.
```
Comment (duglin) [inline docs/sources/reference/commandline/cli.md]: Ditto here too - re: relative.

File: docs/man/docker-cp.1.md (hunk 1 above)
Comment (moxiegirl) [inline docs/man/docker-cp.1.md]: Hi Doug --- (unless it starts with /) --- what does it refer to? "the path in the container" or "HOSTPATH" or something else.  (This isn't really a commentary on the issue...just wanting to learn myself)

Comment (duglin) [inline docs/man/docker-cp.1.md]: @moxiegirl I was thinking of the host running the command.  When I see "paths are relative to the root of the filesystem", I think its fair for someone to think that 'docker cp foo bar' will create /bar on my local system.  When in fact it creates it in my current dir.  But 'docker cp foo /bar' will create /bar.

Comment (duglin) [inline docs/man/docker-cp.1.md]: And yes this text is old, but we might as well fix it at the same time :-)

Comment (ahmetalpbalkan) [inline docs/man/docker-cp.1.md]: Yeah that is totally misleading. I will amend.

Comment (moxiegirl) [inline docs/man/docker-cp.1.md]: Cool. So, I would recommend being more explicit in the text. These sentences are hard to parse because they look simple but are in fact pretty complicated.  ESL people may find them particularly difficult.  (And isn't the opposite of a relative path an absolute one? So, I might mention that for clarity's sake.) I'd suggest:
----
Copy files or folders from a container's filesystem to the Docker host.  The `PATH` is relative to the container.  The `HOSTPATH` is relative to the host's directory in which `docker cp` is run.  For example, if you are running the command in your home directory, `docker cp foo bar` will create `$HOME/bar`  on your host.  To copy to a specific location, specify an absolute pathname for the HOSTPATH parameter.

Comment (ahmetalpbalkan) [inline docs/man/docker-cp.1.md]: >> [...] to the Docker host.

that is the part I want to get rid of.

Comment (moxiegirl) [inline docs/man/docker-cp.1.md]: Was a suggestion. Alter/discard at will. :-D

**Comment by tiborvass** [comment]
Ping @SvenDowideit @moxiegirl @fredlf @jamtur01

**Comment by moxiegirl** [comment]
@tiborvass Did @duglin last query get resolved?  If the answer is yes, LGTM.

**Comment by duglin** [comment]
I still see some "relative" stuff in there

**Comment by jamtur01** [comment]
Yeah ditto

**Comment by tiborvass** [comment]
Sorry guys
ping @ahmetalpbalkan

**Comment by ahmetalpbalkan** [comment]
I still have mixed feelings about this PR.
I probably need somebody native in English here to make the description crystal clear. :smile:

File: docs/man/docker-cp.1.md [hunk 3]
```diff
@@ -11,8 +12,9 @@ CONTAINER:PATH HOSTPATH
...
+the filesystem. Files can be copied from a running or stopped
+container.
```
Comment (duglin) [inline docs/man/docker-cp.1.md]: How about, change:
Paths are relative to the root of the filesystem.
to:
The path within the container is relative to the root of the filesystem.

I don't think we need to talk about the target spot because that will just do what people expect.  We need to mention what's going on in the container because it might not be clear if its relative to / or relative to WORKDIR, or something else.

Comment (moxiegirl) [inline docs/man/docker-cp.1.md]: First, refer to the parameters using the same format as you provide in the synopsis. This eliminates some ambiguity in the sentence structure.  Let the parameters carry the weight instead of the adjective phrases. Then, don't be afraid of providing a definition.  Don't sacrafice clarity to conciseness.

I am a native speaker. :-D

----

Copy files/folders from a `CONTAINER:PATH` to the `HOSTPATH`. The `HOSTPATH` refers to a location on the host running the command. 
The `CONTAINER:PATH` is relative to the root of the container's filesystem. Files can be copied from a running or stopped container.

**Comment by crosbymichael** [comment]
@moxiegirl Could you please take over this change to help @ahmetalpbalkan out?

Thanks all.

**Comment by moxiegirl** [comment]
See #11305  for update of this PR.
